src/NHSD.BuyingCatalogue.Identity.Api/Infrastructure/ValidationSummaryTagHelper.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System.Linq;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+
+namespace NHSD.BuyingCatalogue.Identity.Api.Infrastructure
+{
+    [HtmlTargetElement(TagHelperConstants.Div, Attributes = TagHelperName)]
+    public sealed class ValidationSummaryTagHelper : TagHelper
+    {
+        public const string TagHelperName = "nhs-validation-summary";
+        public const string TitleName = "title";
+
+        [HtmlAttributeNotBound]
+        [ViewContext]
+        public ViewContext ViewContext { get; set; }
+
+        [HtmlAttributeName(TitleName)]
+        public string Title { get; set; }
+
+        public override void Process(TagHelperContext context, TagHelperOutput output)
+        {
+            output.ThrowIfNull();
+            if (ViewContext.ViewData.ModelState.IsValid)
```

Comment:
The property `ViewContext.ViewData.ModelState` is accessed twice, could this be initialised to a variable?

Reply:
With things split into separate methods this doesn't seem like it adds too much value

Reply:
👍